components/TARGET_PSA/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp

```diff
@@ -49,22 +47,66 @@ const uint8_t base64_coding_table[] = {
     '4', '5', '6', '7', '8', '9', '+', '-'
 };
 
-/*
- * \brief Get default KVStore instance for internal flesh storage
- *
- * \return valid pointer to KVStore
- */
-static KVStore *get_kvstore_instance(void)
+static KVStore *kvstore = NULL;
+
+static void its_init(void)
 {
     KVMap &kv_map = KVMap::get_instance();
-
-    KVStore *kvstore = kv_map.get_internal_kv_instance(STR_EXPAND(MBED_CONF_STORAGE_DEFAULT_KV));
+    kvstore = kv_map.get_internal_kv_instance(STR_EXPAND(MBED_CONF_STORAGE_DEFAULT_KV));
     if (!kvstore) {
         // Can only happen due to system misconfiguration.
         // Thus considered as unrecoverable error for runtime.
         error("Failed getting kvstore instance\n");
     }
-    return kvstore;
+
+    its_version_t version = { 0, 0 };
+    size_t actual_size = 0;
+    KVStore::info_t kv_info;
+    int status = kvstore->get_info(ITS_VERSION_KEY, &kv_info);
```

Comment:
is the call to `kvstore->get_info` really required? 
it seems that we can deduce this info from `kvstore->get` return code and actual size.

Reply:
just making sure that no one is trying to fake the size of the key